File: front-end/Mobile/src/models/identity/City.js
import BaseModel from '../BaseModel';

export default class City extends BaseModel {
  constructor(data = {}) {
    super(data);
    this.cityId = data.city_id || null;
    this.name = data.name || '';
    this.department = data.department || null;
  }

  static fromApi(data) {
    if (!data) return null;
    return new City(data);
  }

  toApi() {
    return {
      ...super.toApi(),
      city_id: this.cityId,
      name: this.name,
      department: this.department,
    };
  }
}

File: back-end/01-ms-identity/src/main/java/com/faceattend_edu/identity_service/domain/model/City.java
package com.faceattend_edu.identity_service.domain.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Objects;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class City {
    private Integer cityId;
    private String name;
    private String department;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public void validate() {
        Objects.requireNonNull(this, "City must not be null");
        if (isNullOrBlank(this.name)) throw new IllegalArgumentException("City.name is required");
    }

    private static boolean isNullOrBlank(String str) {
        return str == null || str.trim().isEmpty();
    }
}
